src/js/common/components/RequiredLoginView.jsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { commonStore } from '../store';
+
+import environment from '../helpers/environment.js';
+import { updateLoggedInStatus, updateLogInUrl, updateProfileField } from '../actions';
+
+class RequiredLoginView extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleLogin = this.handleLogin.bind(this);
+    this.handleVerify = this.handleVerify.bind(this);
+    this.setMyToken = this.setMyToken.bind(this);
+    this.getUserData = this.getUserData.bind(this);
+  }
+
+  componentDidMount() {
+    if (localStorage.length > 0) {
```

Comment:
similar to above, what if localStorage has something else in it to make length > 0, but it's not the userToken. Would it be better to check for userToken instead?